internal/backends/rlang/specfile.go

```diff
@@ -0,0 +1,166 @@
+package rlang
+
+import (
+	"os"
+	"io/ioutil"
+	"encoding/json"
+)
+
+type RConfig struct {
+	Packages []RPackage `json:"packages"`
+}
+
+type RPackage struct {
+	Name    string `json:"name"`
+	Version string `json:"version,omitempty"`
+}
+
+func (config RConfig) hasPackage(pkg RPackage) bool {
+	for _, installed := range config.Packages {
+		if installed.Name == pkg.Name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func RAdd(pkg RPackage) {
+	if file, err := os.Open("~/Rconfig.json"); err == nil {
```

Comment:
Same comment as @dan-stowell had on using the working directory. If the config is on the homedir then all the projects will share the same config

Reply:
Make sure you do this for all other files, I see you addressed it only for the `~/R` directory

Reply:
Should be fixed now